code/modules/mob/living/simple_animal/hostile/megafauna/dragon.dm

```diff
@@ -382,7 +382,8 @@
 	..()
 	if(lava_cooldown > world.time)
 		return
-	if(target && istype(target, /turf/open))
+	var/turf/T = get_turf(user)
+	if(target && istype(target, /turf/open) && target in view(7,T))
```

Comment:
wrap the ```target in view(7,T)``` in parenthesis, so:
```... && (target in view(7,T))```

also use user.client.view, not 7, for future proofing.